### PR TITLE
feat: add effective scan config to assetScan annotations

### PR DESCRIPTION
**Pull request by pbalogh-sa**
## Description

This PR adds the effective families configuration as an annotation to the `assetScan` object.
The annotation:
 - key = `openclarity.io/vmclarity-scanner/config`
 - value = the effective configuration in JSON format

closes: #703 

## Type of Change

[ ] Bug Fix  
[x] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: pkg/cli/state/vmclarity_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+)
+
+var effectiveScanConfigJSON = `{"sbom":{"Enabled":true,"AnalyzersList":["syft"],"Inputs":[{"StripPathFromResult":null,"Input":"test","InputType":"dir"}],"MergeWith":null,"AnalyzersConfig":{"Registry":null,"Analyzer":{"OutputFormat":"","AnalyzerList":["syft"],"Scope":"","TrivyConfig":{"Timeout":0,"CacheDir":"","TempDir":""}},"Scanner":null,"LocalImageScan":false}},"vulnerabilities":{"Enabled":false,"ScannersList":null,"Inputs":null,"InputFromSbom":false,"ScannersConfig":null},"secrets":{"Enabled":false,"ScannersList":null,"StripInputPaths":false,"Inputs":null,"ScannersConfig":null},"rootkits":{"Enabled":false,"ScannersList":null,"StripInputPaths":false,"Inputs":null,"ScannersConfig":null},"malware":{"Enabled":false,"ScannersList":null,"StripInputPaths":false,"Inputs":null,"ScannersConfig":null},"misconfiguration":{"enabled":false,"ScannersList":null,"StripInputPaths":false,"Inputs":null,"ScannersConfig":{"Lynis":{"InstallPath":""}}},"infofinder":{"enabled":false,"ScannersList":null,"StripInputPaths":false,"Inputs":null,"ScannersConfig":{"SSHTopology":{}}},"exploits":{"enabled":false,"ScannersList":null,"Inputs":null,"InputFromVuln":false,"ScannersConfig":null}}`
```
Comment (chrisgacsal) [inline pkg/cli/state/vmclarity_test.go:31]: Could you please use `embed` to access test data in tests. Like we did here: https://github.com/openclarity/vmclarity/blob/main/pkg/orchestrator/provider/cloudinit/cloudinit_test.go

**Review by chrisgacsal — CHANGES_REQUESTED** [review]

File: pkg/cli/state/vmclarity_test.go (hunk 1 above)
Comment (pbalogh-sa) [reply pkg/cli/state/vmclarity_test.go]: https://github.com/openclarity/vmclarity/pull/739/commits/35ee574999f8c1d3e31278c25d3ed409f4099feb

File: pkg/cli/state/vmclarity.go [hunk 2]
```diff
@@ -407,3 +415,24 @@ func NewVMClarityState(client *backendclient.BackendClient, id AssetScanID) (*VM
...
+	}
+
+	if annotations == nil {
+		return &effectiveScanConfigAnnotations, nil
+	}
+	newAnnotations := *annotations
+	newAnnotations = append(newAnnotations, effectiveScanConfigAnnotations...)
+
+	return &newAnnotations, nil
+}
```
Comment (Tehsmash) [inline pkg/cli/state/vmclarity.go:438]: We might need to check if the effectiveScanConfigAnnotiationKey is already in the list of annotations, and if it is overwrite it. It shouldn't happen in the normal flow of things but its worth guarding against.

Comment (pbalogh-sa) [reply pkg/cli/state/vmclarity.go:444]: https://github.com/openclarity/vmclarity/pull/739/commits/fab1edc2a4ed52338bc246c3df3fb78d7b897165

File: pkg/cli/state/vmclarity.go [hunk 3]
```diff
@@ -427,11 +427,17 @@ func appendEffectiveScanConfigAnnotation(annotations *models.Annotations, config
...
-	newAnnotations := *annotations
+	var newAnnotations models.Annotations
+	for _, annotation := range *annotations {
+		// If effective scan config annotation exists we will overwrite it.
+		if *annotation.Key == effectiveScanConfigAnnotationKey {
+			continue
+		}
+		newAnnotations = append(newAnnotations, annotation)
+	}
 	newAnnotations = append(newAnnotations, effectiveScanConfigAnnotations...)
```
Comment (Tehsmash) [inline pkg/cli/state/vmclarity.go:441]: I think we can clean this up like:
```
func appendEffectiveScanConfigAnnotation(annotations *models.Annotations, config *families.Config) (*models.Annotations, error) {
	var newAnnotations models.Annotations
	
	if annotatations != nil {
		// Add all odd annotations accept for the effective scan config one if they exist.
		for _, annotation := range *annotations {
			if *annotation.Key == effectiveScanConfigAnnotationKey {
				continue
			}
			newAnnotations = append(newAnnotations, annotation)
		}
	}

    // Add the new effective scan config annotation
	configJSON, err := json.Marshal(config)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal effective families config: %w", err)
	}
	newAnnotations = append(newAnnotations, {
			Key:   utils.PointerTo(effectiveScanConfigAnnotationKey),
			Value: utils.PointerTo(string(configJSON)),
	})
	return &newAnnotations, nil
}
```

Comment (pbalogh-sa) [reply pkg/cli/state/vmclarity.go]: https://github.com/openclarity/vmclarity/pull/739/commits/89fb81ae0783eb6935126ab5c5838945d4e26980

**Review by Tehsmash — APPROVED** [review]

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**